zerver/actions/streams.py

```diff
@@ -294,17 +234,15 @@ def do_unarchive_stream(
 
     recent_traffic = get_streams_traffic({stream.id}, realm)
 
-    # All admins always get to know about private streams' existence,
-    # but we only subscribe the realm owners.
-    send_stream_creation_event(
-        realm, stream, [user.id for user in realm.get_admin_users_and_bots()], recent_traffic
-    )
-    bulk_add_subscriptions(
-        realm=realm,
-        streams=[stream],
-        users=realm.get_human_owner_users(),
-        acting_user=acting_user,
-    )
+    stream_subscribers = get_active_subscriptions_for_stream_id(
+        stream.id, include_deactivated_users=True
+    ).select_related("user_profile")
+    subscribed_users = {sub.user_profile for sub in stream_subscribers}
+    admin_users_and_bots = set(realm.get_admin_users_and_bots())
+
+    notify_users = admin_users_and_bots | subscribed_users
+
+    send_stream_creation_event(realm, stream, [user.id for user in notify_users], recent_traffic)
```

Comment:
Should we only notify admins or subscribers too? I think we should inform subscribers too.

Also do we need to subscribe realm owners? I see no point in that.

Reply:
Also in the case of unarchiving a private stream with no subscribers, what should we do in that case? Not subscribing anyone in stream would make it inaccessible.

Reply:
The [CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/archived.20private.20streams/near/1882647) thread answers some parts of it.